optuna/distributions.py

```diff
@@ -84,9 +84,8 @@ def _asdict(self):
     def __eq__(self, other):
         # type: (Any) -> bool
 
-        if not isinstance(other, type(self)):
-            return False
-
+        if not isinstance(other, BaseDistribution):
```

Comment:
`self` can be a derived class, so we're not directly interested in the type of `self`.